hugo/content/guides/keywords-as-identifiers/_index.md

```diff
@@ -0,0 +1,157 @@
+---
+title: "Keywords as Identifiers"
+weight: 300
+---
+
+When your language uses keywords, which is the usual case, e.g. `var`, `const` or `function` keywords in programming languages, all occurrances of these strings are treated by Langium as keyword tokens by default, even at places which are not intended by the rules in the Langium grammar.
+Additionally, these strings are marked like keywords (in blue in this guide, while other text is white) whenever they are used.
+Summarizing, these keywords cannot be used as values for names, identifiers or other properties by default and need to be explicitly enabled.
+This guide explains, how to do that.
+
+To keep this guide short, let's look at the "hello-world" example, in the [playground](https://langium.org/playground?grammar=OYJwhgthYgBAEgUwDbIPYHU0mQEwFD6IB2ALiAJ6wCyauKAXPrC7ABQAOiIAzmsTwDUAXgAK3PsVgAfWKESJSAS2LAhwgOIgFy1QEoAVAG5C43vyatYAci7ni12MUiJhASQAiJ-Fp0rglqzWSKhojnaSwgDaZpIMngC6NgCE1t4AFkq49FKk3BAqYMiwGADKDLAA9AA6QpUmeSAFzsWeFZVRAPpgALQAXgCCPQBaCVHVAO6dCQb1hJnZJLCNzUU0ADKdAMIA8tTUAKIAcgAq7dU1BuM81aUzAPzVBhdzCznL%2BYXFpZu7%2B8dnKoXC5RAB61WI1RAMzmQA&content=A4UwTgzg9gdgBAKQIYxAKABIgDbaolEAQjTVElji1yjSA) or as a new local project created with `yo langium` (for details, how to set up your first Langium project, read [getting started](/docs/getting-started/)):
+
+![screenshot with the editor, an example and an error message](problem.png)
+
+Here, it is not possible to introduce a person whose name is "Hello", since `Hello` is a dedicated keyword of the language. Additionally, we cannot greet a person called "Hello" as well.
+The same counts for the keyword "person", but in this guide, we focus on enabling "Hello" as name for persons. After reading this guide, you will be able to add support for "person" as name by your own.
+
+To enable keywords as identifiers, you need to apply the following three steps:
+
+1. Tweak the grammar to explicitly parse keywords as property values.
+2. Change the semantic type of the resulting token.
+3. Ensure, that your editor supports styles for the chosen semantic type.
+
+
```

Comment:
Your guide has these 3 parts. It would be nice to introduce headings, so that the reader can skip/fast forward easier.

Reply:
Yes, that would be very helpful, thanks! I added a TOC as well.